azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -733,6 +1039,84 @@ func flattenEventGridEventSubscriptionSubjectFilter(filter *eventgrid.EventSubsc
 	return []interface{}{result}
 }
 
+func flattenEventGridEventSubscriptionAdvancedFilter(input *eventgrid.EventSubscriptionFilter) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
```

Comment:
this also wants to check

```suggestion
	if input == nil || input.AdvancedFilters == nil {
```